app.ts

```diff
@@ -47,8 +59,9 @@ if (process.env.ENABLE_AUTH === 'true') {
     session({
       secret: 'keyboard cat',
       resave: false,
-      saveUninitialized: true,
+      saveUninitialized: false,
```

Comment:
I found a [StackOverflow answer](https://stackoverflow.com/a/40396102/937932) to learn what `saveUninitialized` does, and it suggested a reason why we might want it to be `true`.

> When do you want to enable this? When you want to be able to identify recurring visitors, for example. You'd be able to recognize such a visitor because they send the session cookie containing the unique id.

Maybe we should consider it.  A couple of thoughts to start off:

 - It would enable us to check that Google Analytics calculates a similar number of repeat users
 - We might have to do a privacy impact assessment to ensure that we either aren't storing PII, or we're handling it correctly
 - We aren't using cookies for this, are we?

Reply:
Interesting. It would surely help us calculate the number of repeat users, but only for unauthenticated ones, as far as I understand. Since we're aiming at using SignOn, is that really a use case?

Yes, we are using cookies. Once authenticated, the user sends a cookie to the server with each request. The cookie contains the session ID. Express then checks if the session ID exists in Redis.

Reply:
Good points, thanks.